From daf155fe70c9d69c28bba632b6a758ac8feab6e7 Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@mellanox.com>
Date: Sun, 5 Nov 2017 11:17:28 +0000
Subject: [v4.9 backport 34/35] platform/x86: mlx-platform: fix setting for LED
 masks

It fixes:
 - setting for LED masks: red and green;
 - adds orange LED for next generation systems;

Signed-off-by: Vadim Pasternak <vadimp@mellanox.com>
---
 drivers/leds/leds-mlxreg.c          |  3 ++-
 drivers/platform/x86/mlx-platform.c | 50 ++++++++++++++++++-------------------
 2 files changed, 27 insertions(+), 26 deletions(-)

diff --git a/drivers/leds/leds-mlxreg.c b/drivers/leds/leds-mlxreg.c
index f954463..a932f20 100644
--- a/drivers/leds/leds-mlxreg.c
+++ b/drivers/leds/leds-mlxreg.c
@@ -228,7 +228,8 @@ static int mlxreg_led_config(struct mlxreg_led_priv_data *priv)
 
 		led_cdev = &led_data->led_cdev;
 		led_data->data_parent = priv;
-		if (strstr(data->label, "red")) {
+		if (strstr(data->label, "red") ||
+		    strstr(data->label, "orange")) {
 			brightness = LED_OFF;
 			led_data->base_color = MLXREG_LED_RED_SOLID;
 		} else if (strstr(data->label, "amber")) {
diff --git a/drivers/platform/x86/mlx-platform.c b/drivers/platform/x86/mlx-platform.c
index df10f75..b8fca2a 100644
--- a/drivers/platform/x86/mlx-platform.c
+++ b/drivers/platform/x86/mlx-platform.c
@@ -81,8 +81,8 @@
 #define MLXPLAT_CPLD_PWR_MASK		GENMASK(1, 0)
 #define MLXPLAT_CPLD_FAN_MASK		GENMASK(3, 0)
 #define MLXPLAT_CPLD_FAN_NG_MASK	GENMASK(5, 0)
-#define MLXPLAT_CPLD_LED_LO_NIBBLE_MASK	GENMASK(3, 0)
-#define MLXPLAT_CPLD_LED_HI_NIBBLE_MASK	GENMASK(7, 4)
+#define MLXPLAT_CPLD_LED_LO_NIBBLE_MASK	GENMASK(7, 4)
+#define MLXPLAT_CPLD_LED_HI_NIBBLE_MASK	GENMASK(3, 0)
 
 /* Start channel numbers */
 #define MLXPLAT_CPLD_CH1			2
@@ -589,12 +589,12 @@ static struct mlxreg_core_data mlxplat_mlxcpld_default_led_data[] = {
 	{
 		.label = "psu:green",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED1_OFF,
-		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
 		.label = "psu:red",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED1_OFF,
-		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
 		.label = "fan1:green",
@@ -648,47 +648,47 @@ static struct mlxreg_core_data mlxplat_mlxcpld_msn21xx_led_data[] = {
 	{
 		.label = "status:green",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED1_OFF,
-		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
 		.label = "status:red",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED1_OFF,
-		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK
+		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK
 	},
 	{
 		.label = "fan:green",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED2_OFF,
-		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
 		.label = "fan:red",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED2_OFF,
-		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
 		.label = "psu1:green",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED4_OFF,
-		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
 		.label = "psu1:red",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED4_OFF,
-		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
 		.label = "psu2:green",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED4_OFF,
-		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
 		.label = "psu2:red",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED4_OFF,
-		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
 		.label = "uid:blue",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED5_OFF,
-		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 };
 
@@ -705,19 +705,19 @@ static struct mlxreg_core_data mlxplat_mlxcpld_default_ng_led_data[] = {
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
-		.label = "status:red",
+		.label = "status:orange",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED1_OFF,
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK
 	},
 	{
 		.label = "psu:green",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED1_OFF,
-		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
-		.label = "psu:red",
+		.label = "psu:orange",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED1_OFF,
-		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
+		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
 		.label = "fan1:green",
@@ -725,7 +725,7 @@ static struct mlxreg_core_data mlxplat_mlxcpld_default_ng_led_data[] = {
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
-		.label = "fan1:red",
+		.label = "fan1:orange",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED2_OFF,
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
@@ -735,7 +735,7 @@ static struct mlxreg_core_data mlxplat_mlxcpld_default_ng_led_data[] = {
 		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
-		.label = "fan2:red",
+		.label = "fan2:orange",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED2_OFF,
 		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
@@ -745,7 +745,7 @@ static struct mlxreg_core_data mlxplat_mlxcpld_default_ng_led_data[] = {
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
-		.label = "fan3:red",
+		.label = "fan3:orange",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED3_OFF,
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
@@ -755,7 +755,7 @@ static struct mlxreg_core_data mlxplat_mlxcpld_default_ng_led_data[] = {
 		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
-		.label = "fan4:red",
+		.label = "fan4:orange",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED3_OFF,
 		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
@@ -765,18 +765,18 @@ static struct mlxreg_core_data mlxplat_mlxcpld_default_ng_led_data[] = {
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
-		.label = "fan5:red",
+		.label = "fan5:orange",
 		.reg = MLXPLAT_CPLD_LPC_REG_LED4_OFF,
 		.mask = MLXPLAT_CPLD_LED_LO_NIBBLE_MASK,
 	},
 	{
 		.label = "fan6:green",
-		.reg = MLXPLAT_CPLD_LPC_REG_LED5_OFF,
+		.reg = MLXPLAT_CPLD_LPC_REG_LED4_OFF,
 		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 	{
-		.label = "fan6:red",
-		.reg = MLXPLAT_CPLD_LPC_REG_LED5_OFF,
+		.label = "fan6:orange",
+		.reg = MLXPLAT_CPLD_LPC_REG_LED4_OFF,
 		.mask = MLXPLAT_CPLD_LED_HI_NIBBLE_MASK,
 	},
 };
-- 
2.1.4

